aten/src/ATen/native/Convolution.cpp

```diff
@@ -161,6 +161,8 @@ auto ConvParams::use_cpu_depthwise3x3_winograd(
              (bias.scalar_type() == at::kFloat))) &&
          !is_strided() &&
          !is_dilated() &&
+         // 3x3 depthwith convolutions implementation is inference only
+         !(input.requires_grad() && GradMode::is_enabled()) &&
```

Comment:
I am not familiar with the training side of things, but the question that comes to my mind is what if either input.requires_grad() or GradMode::is_enabled(), but not both, is true.   Say would it be OK if input.requires_grad() is true but GradMode::is_enabled() is false?

Reply:
By the way, there's nothing ARM specific about this code by the looks of it.  Is that OK?  Another consideration is that I vaguely remember privacy preserving team were interested in this code path at one point.  You may want to loop in one of them.

Reply:
@AshkanAliabadi `GradMode::is_enabled()` is a global override, i.e. if it is set to false no gradient computation should happen regardless of whether individual tensor requires grad on not.

And yes, although this code does not look ARM specific, it is gated by `#ifdef __ARM_NEON__` because the op is not implemented for other arches, see:
https://github.com/pytorch/pytorch/blob/5b3c0ae56320fb1289f7525d5f504b51bfd5fec4/aten/src/ATen/native/cpu/DepthwiseConvKernel.cpp#L239-L248